temporal-sdk/src/main/java/io/temporal/internal/client/RootScheduleClientInvoker.java

```diff
@@ -101,12 +101,11 @@ public void createSchedule(CreateScheduleInput input) {
 
     try {
       genericClient.createSchedule(request.build());
-    } catch (Exception e) {
-      if (e instanceof StatusRuntimeException) {
-        StatusRuntimeException sre = (StatusRuntimeException) e;
-        if (Status.Code.ALREADY_EXISTS.equals(sre.getStatus().getCode())) {
-          throw new ScheduleAlreadyRunningException(sre);
-        }
+    } catch (StatusRuntimeException e) {
+      if (Status.Code.ALREADY_EXISTS.equals(e.getStatus().getCode())) {
+        throw new ScheduleAlreadyRunningException(e);
+      } else {
+        throw new ScheduleException(e);
```

Comment:
Wonder if we want to catch every `Exception` then and wrap them all (seems to be what we do with other calls). I also wonder if the `ScheduleException` should extend `WorkflowServiceException` and set `workflowType` to `null` like we do on start workflow. @Quinn-With-Two-Ns - thoughts?

Reply:
>Wonder if we want to catch every Exception then and wrap them all (seems to be what we do with other calls).

Yes we should, I just missed this because we made schedules have its own client

> I also wonder if the ScheduleException should extend WorkflowServiceException and set workflowType to null like we do on start workflow.

I didn't suggest that option because `WorkflowServiceException` extend `WorkflowException` (which is weird, but we can't change that now) and `ScheduleException` is not a `WorkflowException`

>I also wonder if we want it to be ScheduleClientException so people don't think it's for everything about a schedule (but probably not needed). 

I don't like that option because the `ScheduleHandle` should also wrap exceptions so I don't like having "client" in the name.

Reply:
Works for me. Top-level `ScheduleException` that just extends the regular Temporal one as written currently LGTM.